src/main/java/de/unijena/cheminf/fragment/fingerprint/performanceTest/PerformanceTest.java

```diff
@@ -0,0 +1,325 @@
+package de.unijena.cheminf.fragment.fingerprint.performanceTest;
+
+import de.unijena.cheminf.fragment.fingerprint.FragmentFingerprinter;
+import org.openscience.cdk.fingerprint.IBitFingerprint;
+import org.openscience.cdk.fingerprint.ICountFingerprint;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ *  An application for testing the performance of the FragmentFingerprinter methods.
+ *
+ * @author Betuel Sevindik
+ */
+public class PerformanceTest {
+    //<editor-fold defaultstate="collapsed" desc="Private static final constants">
+    /**
+     * Name of file for logging occurred exceptions
+     */
+    private static final String EXCEPTIONS_LOG_FILE_NAME = "Exceptions_Log.txt";
+    /**
+     * Name of file for writing results
+     */
+    private static final String RESULTS_FILE_NAME = "Results";
+    /**
+     * Name of CSV file for writing time processing time
+     */
+    private static final String CSV_TIME_FILE_NAME = "CSV_BIT_PROCESS_TIME";
+    /**
+     * Name of CSV file for the network SMILES and the number of origins
+     */
+    private static final String CSV_ORIGIN_NETWORK_FILE_NAME = "CSV_COUNT_PROCESS_TIME";
+    /**
+     * Name of CSV file for the forest SMILES and the number of origins
+     */
+    private static final String CSV_ORIGIN_FOREST_FILE_NAME = "CSV_Origin_Forest";
+    //</editor-fold>
+    //
+    //<editor-fold defaultstate="collapsed" desc="Private class variables">
+    /**
+     * The working directory (the jar-file's directory)
+     */
+    private String workingPath;
+    /**
+     *  List which is contains all bit fingerprints
+     */
+    private ArrayList<ArrayList<String>> moleculeListforBitFingerprint = new ArrayList<>();
```

Comment:
Please initialize ArrayLists with an initial size, big enough to store the expected amount of entries